src/shared-components/segmentedControl/index.tsx

```diff
@@ -42,11 +38,11 @@ export const SegmentedControl: React.FC<SegmentedControlProps> = ({
       setTargetWidth(targetRef.current.offsetWidth);
       setTransform(
         `translate3d(${
-          targetRef.current.offsetWidth * activeSegmentIndex
+          targetRef.current.offsetWidth * initialActiveIndex
         }px, 0, 0)`,
       );
     }
-  }, [targetRef]);
+  }, [initialActiveIndex]);
```

Comment:
`initialActiveIndex` captures that this is an effect that runs `onMount`, all other changes after rendering are handled by `onSegmentClick`, not with side-effects.